docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,271 @@
+   Specifies the client secret assigned by the OIDC provider to the API ML Gateway. It is used in combination with the client_id in Access Token validation requests at the /introspect endpoint of the OIDC provider.
+
+   * **`components.gateway.apiml.security.oidc.registry`**  
+   Specifies the trusted registry name used by SAF to group the identities recognized as having a OIDC identity mapping. 
+   The trusted registry name is the string used during the creation of the mapping between the distributed and mainframe user identities. See the [ESM configuration](#esm-configuration) for more information.
+
+   * **`components.gateway.apiml.security.oidc.introspectEndpoint`**
+   Specifies the path portion of the token introspection endpoint URL, provided by the OAuth2/OIDC provider. The Token Introspection endpoint is defined by the [OAuth 2.0 Token Introspection extension](https://datatracker.ietf.org/doc/html/rfc7662) as an OAuth2 endpoint, that takes a parameter representing an OAuth 2.0 token and returns a JSON document representing the meta information surrounding the token, including whether this token is currently active.
+
+
+**Note:** Provide externalMapperUser and externalMapperUrl if they differ from the default values.     
+   
+  * **'components.gateway.apiml.security.x509.externalMapperUser'**
+    The user ID used to call the external mapper. Needs to be set properly according to the permissions requirements listed in the step 2 of the [ESM configuration](#esm-configuration) above.   
+
+  **Note:** Skip this property if the Zowe runtime userId is not altered from the default `ZWESVUSR`.
+
+To authenticate to the mapping API, a JWT is sent with the request. The token represents the user that is configured with this property. The user authorization is required to use the `IRR.RUSERMAP` resource within the `FACILITY` class. The default value is `ZWESVUSR`. Permissions are set up during installation with the `ZWESECUR` JCL or workflow.
+
+If you customized the `ZWESECUR` JCL or workflow (the customization of zowe runtime user: `// SET ZOWEUSER=ZWESVUSR * userid for Zowe started task`) and changed the default USERID, create the `components.gateway.apiml.security.x509.externalMapperUser` property and set the value by adding a new line as in the following example:
+
+**Example:**
+
+   ```
+   components.gateway.apiml.security.x509.externalMapperUser: yournewuserid  
+   ```
+
+   * **`apiml.security.x509.externalMapperUrl`**  
+  Defines the URL where Gateway can query the mapping of distributed user ID to mainframe user ID. 
+  This property informs the Gateway about the location of this API. ZSS is the default API provider in Zowe. You can provide your own API to perform the mapping. In this case, it is necessary to customize this value.
+
+The following URL is the default value for Zowe and ZSS:
+
+     ```
+     https://${ZWE_haInstance_hostname}:${GATEWAY_PORT}/zss/api/v1/certificate/x509/map
```

Comment:
default value is `https://${ZWE_haInstance_hostname}:${GATEWAY_PORT}/zss/api/v1/certificate/dn`